src/plugins/intel_cpu/src/nodes/lora.h

```diff
@@ -21,18 +21,16 @@ class LoRA : public Node {
         return getType() == Type::LoRA;
     }
 
-    bool needPrepareParams() const override {
```

Comment:
should not we override "needShapeInfer" instead and use some specific shapeInder method which would call "updateShapes" for the input nodes of the inner graph?

Reply:
Unfortunately updateShapes call doesn't refresh the `lastInputDims`. This data are updated only after the `executeDynamic` call (as it is used by the `needPrepareParams` call as well). So calling `updateShapes` to the input nodes won't eventually trigger the `prepairParams` if there is shape1->emptyShape->shape1 sequence (which triggered the original bug). Forcing `prepareParams` for inputs each exec call will end up in a substantial performance penalty.

Reply:
I understand that currently we update lastInputDims only after the node execution finished, which is fine generalization. But, in general, this is only really necessary for the nodes with the internal dynamism. Most of the nodes can update its lastInputDims after updateShapes() is finished.
What do you think?

Reply:
Nodes with internal dynamism aren't the reason of updating `lastInputDims` in the exec call, as the internal dynamism only means the moment of deducing the _output_ shapes not input. Please note that `needPrepareParams` also checks if the shape is different from the previous one. Thus, if we update `lastInputDims` in `updateShapes()`, then `needPrepareParams` will be false as the current input shapes are compared with the already updated `lastInputDims`. Therefore, in order to change the moment of updating `lastInputDims` we will have to refactor the `needPrapareParams` check as well.